containers/Dockerfile.buildenv

```diff
@@ -72,46 +76,59 @@ RUN apt-get update -y \
     && printf '. /etc/profile.d/lmod.sh\n' >> /root/.bashrc \
     && . /etc/profile.d/lmod.sh
 
-# Install Spack to get remaining dependencies
-WORKDIR /work
-RUN git clone https://github.com/LLNL/spack.git
-WORKDIR /work/spack
-# Since spack/develop is rather unstable, we check out a commit we
-# know is stable. This should be updated periodically to update
-# installed packages.
-RUN git checkout 470a45c51659156e7d154ea890e798ce32b8767d
-WORKDIR /work
-
-# Spack needs to be pointed to the system OpenSSL to work properly, we add this
-# in the general configure script for Spack rather than a user-specific
-# configure script. The below code is documented in the Spack manual.
-RUN printf "\n  openssl:\n    paths:\n      openssl@1.0.2g: /usr\n\
-    buildable: False\n" >> /work/spack/etc/spack/defaults/packages.yaml
-
-# Add Spack to PATH and install required dependencies
-# The sed commands are necessary because spack fails to find the
-# fortran compilers for an unknown reason. We compile the libraries with GCC6
-# so that we do not need separate versions for GCC6 and GCC7.
-RUN echo "export PATH=\$PATH:/work/spack/bin" >> /root/.bashrc\
-    && echo '. /work/spack/share/spack/setup-env.sh' >> /root/.bashrc \
-    && export PATH=$PATH:/work/spack/bin \
-    && spack compiler find \
-    && sed -i 's@fc: null@fc: /usr/bin/gfortran@' \
-    /root/.spack/linux/compilers.yaml \
-    && sed -i 's@f77: null@f77: /usr/bin/gfortran@' \
-    /root/.spack/linux/compilers.yaml
-RUN /work/spack/bin/spack install cmake \
-    && /work/spack/bin/spack install --no-checksum catch@2.1.0 \
-    && /work/spack/bin/spack install brigand@master \
-    && /work/spack/bin/spack install blaze \
-    && /work/spack/bin/spack install gsl%gcc@6.5.0 \
-    && /work/spack/bin/spack install libsharp -openmp -mpi \
-    && /work/spack/bin/spack install libxsmm%gcc@6.5.0 \
-    && /work/spack/bin/spack install yaml-cpp@develop%gcc@6.5.0 \
-    && /work/spack/bin/spack install benchmark%gcc@6.5.0
+# We install dependencies not available through apt manually rather than using
+# Spack since Spack ends up building a lot of dependencies from scratch
+# that we don't need. Thus, not building the deps with Spack reduces total
+# build time of the Docker image.
```

Comment:
👍 for this comment. Relatedly: now that we aren't using Spack in the container, do we still need to install LMod (new line 70ish, not part of this PR)?

Reply:
Yes, I was partly too lazy to wait for the image to build again :P I'll remove once the other comments are settled :)